### PR TITLE
Status change log

### PR DESCRIPTION
**Pull request by aviskarkc10**

### CODE REVIEW

File: src/components/dashboard/LogList.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import LogRow from './LogRow';
+
+const LogList = ({ logs }) => (
+
```
Comment (pabin2) [inline src/components/dashboard/LogList.js]: Extra line

File: src/components/dashboard/LogRow.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import ToolTip from 'react-tooltip';
+
+const LogRow = ({ log }) => {
+  let { name } = log.service;
+  let { createdAt } = log;
+  let status = log.status.name;
+  let statusMessage = `Went ${status} at ${createdAt}`
+  let className = "log-status-"+status;
```
Comment (pabin2) [inline src/components/dashboard/LogRow.js]: Better not to use reserve keyword as a string . (className)

File: src/components/dashboard/StatusPanel.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ import * as statusService from '../../services/status';
 
 import Panel from '../commons/Panel';
 import ServiceList from './ServiceList';
+import LogList from './LogList';
+
```
Comment (pabin2) [inline src/components/dashboard/StatusPanel.js]: Extra line . And make sure you add imported files in proper order . 
.
..
...
....


```
import LogList from './LogList';
import ServiceList from './ServiceList';
import Panel from '../commons/Panel';
```

File: src/components/dashboard/LogList.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import LogRow from './LogRow';
```
Comment (pabin2) [inline src/components/dashboard/LogList.js]: Good to have a new line between installed 3rd party package and our imported Components .

File: src/components/dashboard/StatusPanel.js [hunk 5]
```diff
@@ -48,9 +52,19 @@ class StatusPanel extends Component {
     }
 
     return (
-      <Panel title={message} className={className}>
-        <ServiceList services={services} />
-      </Panel >
+
+      <div>
+
```
Comment (pabin2) [inline src/components/dashboard/StatusPanel.js]: Formatting issue . (extra lines)

File: src/index.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ const mountNode = document.getElementById('app');
 document.title = config.app.title;
 
 const renderApp = () => {
+
```
Comment (pabin2) [inline src/index.js]: here

File: webpack.config.js [hunk 7]
```diff
@@ -79,6 +80,8 @@ async function configureWebpack() {
       new webpack.HotModuleReplacementPlugin() // enable HMR globally
     ]
   };
+  });
+
```
Comment (pabin2) [inline webpack.config.js]: Here

**Comment by pabin2** [comment]
Hi @aviskarkc10 ,
Instead of past tense we use present tense to name our commits.
Example 
added log for status change
would have been
Add logs for status change.

File: src/components/dashboard/LogRow.js (hunk 2 above)
Comment (sanjevShakya) [inline src/components/dashboard/LogRow.js]:
```javascript
let { service, createdAt, status } = log;
let statusName = status.name;
```

File: src/components/dashboard/ServiceList.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ import ServiceRow from './ServiceRow';
 const ServiceList = ({ services }) => (
   <ul className="list-group">
     {
-      services.map(service => <ServiceRow service={service} key={service.id} />)
+     	services.map(service => <ServiceRow service={service} key={service.id} />)
```
Comment (sanjevShakya) [inline src/components/dashboard/ServiceList.js]:
```javascript
services && services.map(service => .....)
```

File: src/components/hoc/status.js [hunk 9]
```diff
@@ -12,6 +12,7 @@ export const withStatusInfo = WrappedComponent => {
       this.state = {
         status: {
           services: [],
+          logs:[],
```
Comment (sanjevShakya) [inline src/components/hoc/status.js]: Can you please apply length rule here as well.

File: public/css/main.css [hunk 10]
```diff
@@ -73,6 +73,14 @@ a:hover {
   background: #fff;  
 }
 
+.log-status-Up {
+  color: #19be10;
+}
+
+.log-status-Down{
```
Comment (sanjevShakya) [inline public/css/main.css]: Can you please change the css naming to .log-status-up. naming consistency.
```
.log-status-up {
...
}
.log-status-down {
...
}
```

**Comment by kabirbaidhya** [comment]
@aviskarkc10 @aishwaryashrestha11 Also, go through these [contributing guidelines](https://github.com/leapfrogtechnology/chill/blob/master/CONTRIBUTING.md), this roughly describes our conventions. 
And make sure you do `yarn lint:fix` before you commit. You can identify issues before hand.

File: src/components/dashboard/StatusPanel.js [hunk 11]
```diff
@@ -25,20 +26,22 @@ class StatusPanel extends Component {
...
-
-    updateStatus({ isLoading: true, services: [] });
+    updateStatus({ isLoading: true, services: [], logs: [] });
 
     try {
       let services = await statusService.fetchServiceStatuses();
-
-      updateStatus({ services, isLoading: false });
+      
+      //fetch logs
```
Comment (kabirbaidhya) [inline src/components/dashboard/StatusPanel.js]:
```js
// Fetch logs
```
One space after `//` is required for cleanliness.

File: src/config.js [hunk 12]
```diff
@@ -11,6 +11,12 @@ export default {
     },
     baseUrl: __INJECTED_CONFIG.dashboard.apiBaseUrl || 'http://localhost:8000/api'
   },
+  logApi: {
```
Comment (kabirbaidhya) [inline src/config.js]: No need another wrapper `logApi` just for a new endpoint. Just append it under `api.endpoints` like this:

```js
   api: {
     endpoints: {
       status: '/status',
       statusLogs: '/status/logs'
      },
      baseUrl: __INJECTED_CONFIG.dashboard.apiBaseUrl || 'http://localhost:8000/api'
    },
```

File: src/components/dashboard/LogList.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import LogRow from './LogRow';
+
+const LogList = ({ logs }) => (
+	<ul className="list-group">
+	{
+      logs.map(log => <LogRow log={log} key={log.id} />)
```
Comment (kabirbaidhya) [inline src/components/dashboard/LogList.js]: Inconsistent indentation. Re-format here.

File: src/components/dashboard/LogRow.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const LogRow = ({ log }) => {
+  let { name } = log.service;
+  let { createdAt } = log;
+  let status = log.status.name;
+  let statusMessage = `Went ${status} at ${createdAt}`;
+  let textColor = 'log-status-' + status.toLowerCase();
```
Comment (romitamgai) [inline src/components/dashboard/LogRow.js]: use es6 string literal

**Pull request merged**